internal/cli/command_factory.go

```diff
@@ -106,6 +107,13 @@ func (factory *CommandFactory) Build(command CommandDefinition) *cobra.Command {
 			factory.telemetryService.TrackEvent(telemetry.EventTypeCommandStart)
 
 			err := command.Command.Handler(factory.profile, factory.ui)
+			if err == realm.ErrInvalidSession {
+				factory.profile.ClearSession()
+				profileErr := factory.profile.Save()
+				if profileErr != nil {
+					factory.errLogger.Fatal(fmt.Errorf("failed to clear session: %s", profileErr.Error()))
```

Comment:
I wasn't sure of the _best_ spot to save the profile, but I placed it above the error check and added a log in case the profile does file to save for whatever reason. I imagine like we can reduce the log level too, considering that this isn't a big issue. Maybe an info?

Reply:
I think this is fine here, we shouldn't be making any api calls from the `Setup` or `Feedback` stages

and yeah, we don't want to use `errLogger.Fatal` for a failure to save the profile after attempting to clear the session.  the error that still shows to the user is "invalid session" and they'll still be prompted to re-login (effectively clearing their session at that point).

in the interest of not having unhandled errors, and considering we're in the `RunE` phase which means we know we've ensured our `ui` exists, we should just use that to print a text message with a WARN severity (you'll have to add that severity)